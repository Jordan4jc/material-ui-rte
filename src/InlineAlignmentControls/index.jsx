import React from 'react'
import { withStyles } from "@material-ui/styles"
import Button from '@material-ui/core/Button'
import ButtonGroup from '@material-ui/core/ButtonGroup'
import FormatAlignLeftIcon from "@material-ui/icons/FormatAlignLeft"
import FormatAlignCenterIcon from "@material-ui/icons/FormatAlignCenter"
import FormatAlignRightIcon from "@material-ui/icons/FormatAlignRight"
import clsx from 'clsx';
import styles from "./styles"

const INLINE_STYLES = [
  {icon: FormatAlignLeftIcon, style: 'ALIGN-LEFT'},
  {icon: FormatAlignCenterIcon, style: 'ALIGN-CENTER'},
  {icon: FormatAlignRightIcon, style: 'ALIGN-RIGHT'}
];

const InlineAlignmentControls = (props) => {
  const { classes } = props
  const currentStyle = props.editorState.getCurrentInlineStyle();
  const renderIcon = type => {
    let IconName = type.icon
    return <IconName className={clsx({
      [classes.activeIcon]: currentStyle.has(type.style)
    })} />
  }
  return (
    <ButtonGroup size="small" variant="contained">
      {INLINE_STYLES.map((type) =>
        <Button key={type.style} onClick={() => props.onChange(type.style)}>{renderIcon(type)}</Button>
        // <Button key={type.style} onClick={() => console.log(props.editorState.getBlockTree())}>{renderIcon(type)}</Button>
      )}
    </ButtonGroup>
  );
};

export default withStyles(styles)(InlineAlignmentControls)