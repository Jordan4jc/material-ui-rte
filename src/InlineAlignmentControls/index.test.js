import React from "react"
import { shallow } from "enzyme"
import { EditorState } from "draft-js"
import { unwrap } from "@material-ui/core/test-utils";
import InlineAlignmentControls from "./index"

const InlineAlignmentControlsUnwrapped = unwrap(InlineAlignmentControls);

describe("InlineAlignmentControls", () => {
  let editorState

  beforeEach(() => {
    editorState= EditorState.createEmpty()
  })

  it("Should render alignment control", () => {
    const component = shallow(<InlineAlignmentControlsUnwrapped editorState={editorState} classes={{}} />);
  
    expect(component).toMatchSnapshot();
  });
});