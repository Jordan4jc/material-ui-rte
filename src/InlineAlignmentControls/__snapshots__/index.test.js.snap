// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineAlignmentControls Should render alignment control 1`] = `
<WithStyles(ForwardRef(ButtonGroup))
  size="small"
  variant="contained"
>
  <WithStyles(ForwardRef(Button))
    key="ALIGN-LEFT"
    onClick={[Function]}
  >
    <FormatAlignLeftIcon
      className=""
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    key="ALIGN-CENTER"
    onClick={[Function]}
  >
    <FormatAlignCenterIcon
      className=""
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    key="ALIGN-RIGHT"
    onClick={[Function]}
  >
    <FormatAlignRightIcon
      className=""
    />
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(ButtonGroup))>
`;
